plugins/item_tasks/server/rest.py

```diff
@@ -320,3 +323,94 @@ def setSpecFromXml(self, item, xml, setName, setDescription, params):
             'itemTaskSpec': itemTaskSpec,
             'isItemTask': True
         })
+
+    @access.admin(scope=constants.TOKEN_SCOPE_AUTO_CREATE_CLI)
+    @autoDescribeRoute(
+        Description('Create item task specs based on a docker image.')
+        .notes('This operates on an existing folder, adding item tasks '
+               'using introspection of a docker image.')
+        .modelParam('id', 'The ID of the folder that the task specs will be added to.',
+                    model='folder', level=AccessType.WRITE)
+        .param('image', 'The docker image name.', required=True, strip=True)
+        .param('pullImage', 'Whether the image should be pulled from a docker registry. ' +
+               'Set to false to use local images only.',
+               dataType='boolean', required=False, default=True)
+    )
+    def runJsonTasksDescription(self, folder, image, pullImage, params):
+        jobModel = self.model('job', 'jobs')
+        token = self.model('token').createToken(
+            days=3, scope='item_task.set_task_spec.%s' % folder['_id'],
+            user=self.getCurrentUser())
+        job = jobModel.createJob(
+            title='Read docker task specs: %s' % image, type='item_task.json_description',
+            handler='worker_handler', user=self.getCurrentUser())
+
+        jobOptions = {
+            'itemTaskId': folder['_id'],
+            'kwargs': {
+                'task': {
+                    'mode': 'docker',
+                    'docker_image': image,
+                    'container_args': [],
+                    'pull_image': pullImage,
+                    'outputs': [{
+                        'id': '_stdout',
+                        'format': 'text'
+                    }],
+                },
+                'outputs': {
+                    '_stdout': {
+                        'mode': 'http',
+                        'method': 'PUT',
+                        'format': 'text',
+                        'url': '/'.join((getApiUrl(), self.resourceName,
+                                         str(folder['_id']), 'add_json_specs')),
+                        'headers': {'Girder-Token': token['_id']},
+                        'params': {
+                            'image': image,
+                            'pullImage': pullImage
+                        }
+                    }
+                },
+                'jobInfo': utils.jobInfoSpec(job),
+                'validate': False,
+                'auto_convert': False,
+                'cleanup': True
+            }
+        }
+        job.update(jobOptions)
+
+        job = jobModel.save(job)
+        jobModel.scheduleJob(job)
+        return job
+
+    @access.token
+    @autoDescribeRoute(
+        Description('Create item tasks under a folder using a list of JSON specifications.')
+        .modelParam('id', model='folder', force=True)
+        .jsonParam('json', 'The JSON specs.', paramType='body')
```

Comment:
I have not done this, since this schema seemed somewhat complicated (can be array or single object). If you know how to put a basic check like this in I'd welcome that, otherwise we may want to add item task schema specification on another PR.